tests/projects/test_projects.py

```diff
@@ -285,11 +285,12 @@ def test_run_async():
         ),
     ],
 )
-def test_conda_path(mock_env, expected_conda, expected_activate):
+def test_conda_path(mock_env, expected_conda, expected_activate, monkeypatch):
     """Verify that we correctly determine the path to conda executables"""
-    with mock.patch.dict("os.environ", mock_env, clear=True):
-        assert mlflow.utils.conda.get_conda_bin_executable("conda") == expected_conda
-        assert mlflow.utils.conda.get_conda_bin_executable("activate") == expected_activate
+    for k, v in mock_env.items():
+        monkeypatch.setenv(k, v)
+    assert mlflow.utils.conda.get_conda_bin_executable("conda") == expected_conda
+    assert mlflow.utils.conda.get_conda_bin_executable("activate") == expected_activate
```

Comment:
the test is failing now. we need to clear the `MLFLOW_CONDA_HOME` and `CONDA_EXE` env at the start of this test

Reply:
Do you think I should use `delenvs`?

Reply:
that should work :)